pkg/kube/cluster-init.sh

```diff
@@ -137,7 +179,9 @@ else
                 logmsg "Starting k3s server after reboot"
                 nohup /usr/bin/k3s server --config /etc/rancher/k3s/config.yaml &
                 logmsg "Looping until k3s is ready"
-                until kubectl get node | grep "$HOSTNAME" | awk '{print $2}' | grep 'Ready'; do sleep 5; done
+                until kubectl get node | grep "$(/bin/hostname)" | awk '{print $2}' | grep 'Ready'; do sleep 5; done
```

Comment:
Can we simplify this by doing the following?

```sh
HOSTNAME=$(/bin/hhostname)
until kubectl get node ${HOSTNAME}  --show-labels=false | grep -w Ready; do sleep 5; done
```

That should avoid constantly calling out to `/bin/hostname`, skip some `awk` calls, ensure it only looks at the current node, and also not trick you by accepting `NotReady` (`grep -w Ready` fixes that).

Reply:
Yeah this to avoid waiting forever for a node to come ready with the name 'linuxkit*' in the case where hostname changes from linuxkit* to <uuid> while k3s is starting up.

Reply:
So it changes? Hence looking for `${HOSTNAME}` will not work, as `${HOSTNAME}` is only _part_ of it? Hence the grep? Understood.

Reply:
It still is somewhat unpredictable. Are we not able to determine, "this is what the node name will be" and search for it? Neither k3s nor eve just sets up completely random hostnames.

Reply:
@deitch EVE uses the device uuid as hostname (to make it be unique - something which was required back when we ran LISP overlay networks). Thus before the device has been onboarded it does not know its uuid=hostname.  Once onboarded to the controller it sets the hostname early during boot but it makes sense for the init scripts to wait for it to be known. 
We have a way of doing that for golang code (using pubsub waiting for the OnboardingStatus), but no easy way in shell scripts.

Reply:
OK, I can see that. Then maybe `grep -w` so it is consistent? The node name _does_ match the hostname? Or even just `kubectl get node $(hostname)`? Or is the node name different from the hostname somehow?

Reply:
@eriknordmark @deitch  Yes the node name is the hostname.  And good point we should be able to condense and remove the awk.  kubectl also has a json output and I suppose we could cut another pipe and grep with this below. 

`42b04062-e53d-4c17-bbeb-9dc4e6b61f6b:/# kubectl get node 42b04062-e53d-4c17-bbeb-9dc4e6b61f6b -o json | jq '.status.conditions[] | select(.reason=="KubeletReady") | .status=="True"'
true`

Reply:
Yeah, that is much more predictable and clean. No chance of something getting messy with the `grep`. 

Also, I am impressed with your jq-foo.

Reply:
Thanks! @deitch